src/pages/workspace/accounting/qbd/QuickBooksDesktopSetupPage.tsx

```diff
@@ -71,9 +92,28 @@ function RequireQuickBooksDesktopModal({route}: RequireQuickBooksDesktopModalPro
                 onBackButtonPress={() => Navigation.dismissModal()}
             />
             <ContentWrapper>
-                {isLoading || !codatSetupLink ? (
-                    <FullScreenLoadingIndicator style={[styles.flex1, styles.pRelative]} />
-                ) : (
+                {shouldShowLoading && <FullScreenLoadingIndicator style={[styles.flex1, styles.pRelative]} />}
```

Comment:
Looks like handling loading via `FullScreenLoadingIndicator` led to this issue:
- https://github.com/Expensify/App/issues/53608

which we agreed on fixing by showing the setup link UI right away when RHP opens and displaying an `ActivityIndicator` in the link box while `hasResultOfFetchingSetupLink` is being fetched.